### PR TITLE
Connections: Track current chat page

### PR DESCRIPTION
**Pull request by mia-pi-git**
Needed for #7333.

### CODE REVIEW

**Comment by Zarel** [comment]
Didn't we agree to have a `Set` of open pages? Whatever happened to that?

**Comment by mia-pi-git** [comment]
> Didn't we agree to have a `Set` of open pages? Whatever happened to that?

I don't remember that, but i can certainly do it.

**Comment by mia-pi-git** [comment]
Done, @Zarel.

File: server/chat-commands/moderation.ts [hunk 1]
```diff
@@ -422,7 +422,10 @@ export const commands: ChatCommands = {
 	part(target, room, user, connection) {
 		const targetRoom = target ? Rooms.search(target) : room;
 		if (!targetRoom) {
-			if (target.startsWith('view-')) return;
+			if (target.startsWith('view-')) {
+				connection.currentPages?.delete(target.slice(5));
```
Comment (Zarel) [inline server/chat-commands/moderation.ts:426]:
```suggestion
				connection.currentPages?.delete(target.slice(5));
				if (!connection.currentPages?.size()) connection.currentPages = null;
```

File: server/chat-commands/moderation.ts [hunk 2]
```diff
@@ -422,7 +422,11 @@ export const commands: ChatCommands = {
...
+				if (!connection.currentPages?.size()) connection.currentPages = null;
```
Comment (Zarel) [inline server/chat-commands/moderation.ts:427]:
```suggestion
				if (!connection.currentPages?.size) connection.currentPages = null;
```

Ugh, I never remember if `set.size` is a function.

**Comment by Zarel** [comment]
Can you rename from `currentPages` to `openPages`?

**Comment by Zarel** [comment]
Looks good.

**Pull request merged**